### PR TITLE
Add an adventure text colour parser

### PR DESCRIPTION
**Pull request by Sauilitired**
**Untested** Should probably be tested before merging.

### CODE REVIEW

File: cloud-minecraft/cloud-minecraft-extras/src/main/java/cloud/commandframework/minecraft/extras/ColourArgument.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+        @Override
+        public @NonNull List<@NonNull String> suggestions(
+                final @NonNull CommandContext<C> commandContext, final @NonNull String input
+        ) {
+            final List<String> suggestions = new LinkedList<>();
+            if (input.isEmpty() || input.equals("#") || (HEX_PREDICATE.test(input) && input.length() < 7)) {
+                for (char c = 'a'; c <= 'f'; c++) {
+                    suggestions.add(String.format("%s%c", input, c));
+                    suggestions.add(String.format("&s%c", c));
```
Comment (broccolai) [inline cloud-minecraft/cloud-minecraft-extras/src/main/java/cloud/commandframework/minecraft/extras/ColourArgument.java:200]:
```suggestion
                    suggestions.add(String.format("&%c", c));
```

File: cloud-minecraft/cloud-minecraft-extras/src/main/java/cloud/commandframework/minecraft/extras/ColourArgument.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+                }
+                for (char c = '0'; c <= '9'; c++) {
+                    suggestions.add(String.format("%s%c", input, c));
+                    suggestions.add(String.format("&s%c", c));
```
Comment (broccolai) [inline cloud-minecraft/cloud-minecraft-extras/src/main/java/cloud/commandframework/minecraft/extras/ColourArgument.java:204]:
```suggestion
                    suggestions.add(String.format("&%c", c));
```

**Review by broccolai — CHANGES_REQUESTED** [review]
Suggestions still appear red when going onto the next one

**Comment by jmanpenilla** [comment]
Maybe we could have multiple modes for the parser.
One mode similar to how it is now, and a separate brig-friendly mode which will only suggest NamedTextColors but will accept a NamedTextColor or a 6-char hex.

**Review by jmanpenilla — APPROVED** [review]

**Pull request merged**